Non

seifoh opened the pull request:
git remote add origin https://github.com/seifoh/seifoo.git
git push -u origin master

osm0sis committed:
Bringup to busybox 1.24.1 with updated configs for without patches (basic) and with (full)

osm0sis committed:
A couple more fixups to docs/configs before introducing/fixing patches

osm0sis committed:
Patches: Fix build warnings on Cygwin and add custom build tag

osm0sis committed:
patch: hack the hack to make mount, etc. work again with the newer NDK

osm0sis committed:
fix: rebase ash-history patch for busybox 1.24.1

osm0sis committed:
patches: telnet/telnetd and rfkill, add missing headers

osm0sis committed:
More repo updates and cleanups
- Update README for upstreamed patches no longer needed as of 8488d53c63126488c88df78623b9123a61da4d03
- Update configs to reflect new default options
- Remove outdated Makefile and update instructions, label old configs more clearly

osm0sis committed:
Readme again, typos and formatting

osm0sis committed:
fix: rename to group dependent patches together

osm0sis committed:
fix: rebase part 1 of arping/udhcpd/ether patches for busybox 1.24.1

osm0sis committed:
fix: finish renaming to group dependent ether patches together

osm0sis committed:
fix: rebase part 2 of arping/udhcpd/ether patches for busybox 1.24.1

osm0sis committed:
patch: hack the hack for arping/ether/udhcpd

osm0sis committed:
patch: loadfont/setfont/conspy, add missing header

osm0sis committed:
update full config and README to reflect latest patches

osm0sis committed:
fix: rebase swapon-swapoff patch for busybox 1.24.1

osm0sis committed:
fix: rename to group dependent ipc*/syslogd/logread patches together

osm0sis committed:
fix: rename to group dependent swap patches together

osm0sis committed:
configs/README update

osm0sis committed:
fix: rebase misc mod* fixes (not required to build) for busybox 1.24.1

osm0sis committed:
patches: eject, ifenslave, ifplugd, and feature_ifconfig_slip; add missing headers

osm0sis committed:
config/README for latest fixes

osm0sis committed:
patches: fix more build warnings on Cygwin when using menuconfig

osm0sis committed:
fix up configs one more time
- match menuconfig output and add full modutils by default to both
- enable a few more options from the defconfig on full
- add -fno-builtin-stpcpy to remove warnings on x86 compiles

osm0sis committed:
patch: update busybox's default setup for Android API 21+

osm0sis committed:
patch: nandwrite/nanddump, add missing headers (API 21 only)

osm0sis committed:
add configs for 64bit/API 21, previous were 32bit/API 9

osm0sis committed:
fix: correct mntent hack to make df, etc. work again

osm0sis committed:
config/README fixes and updates
- enable SYNC since it actually only has linker problems when FEATURE_SYNC_FANCY is enabled
- enable FEATURE_LS_COLOR because Titanium Backup is weirdly dependent on it to generate its app lists
- disable FEATURE_XARGS_SUPPORT_REPL_STR because Titanium Backup requires the expected output for batch processing

osm0sis committed:
Bump to 1.24.2
- all continues to compile correctly with my changes

osm0sis committed:
config: enable swaponoff for Android x64 full
- turns out it was dependent on hasmntopt from 003-a (which is otherwise unnecessary now that mount, etc. compile correctly with API 21)

osm0sis committed:
Revert "patch: update busybox's default setup for Android API 21+"
- upstreamed in busybox 1.25.0

This reverts commit 484b84536d486902ab8e282328210c983bd54611.

osm0sis committed:
Revert "patch: hack the hack for arping/ether/udhcpd"
- upstreamed in busybox 1.25.0

This reverts commit 7f93b2adb6d290f98888dc610ffcae22d014f7be.

osm0sis committed:
fixes: rebase syscalls patches for busybox 1.25.0
- adjust syscalls patches broken by official version of upstreamed 002-platform-missing_syscalls

osm0sis committed:
fix: rebase modinfo-modprobe-without-utsrel patch for busybox 1.25.0

osm0sis committed:
fix: rebase depmod-parameter patch for busybox 1.25.0

osm0sis committed:
patch: blkdiscard, add missing defines (API <21)

osm0sis committed:
fix: rebase arping/ether/udhcpd+minix for busybox 1.25.0
- ether_ntoa is not used in arping anymore
- mempcpy is handled correctly internally

osm0sis committed:
patch: ipneigh, do not include neighbour.h on Android

osm0sis committed:
Bringup to busybox 1.25.0 with updated configs
- ipneigh patch is only necessary on Android API <21 since neighbour.h is now properly included in the NDK
- similarly, blkdiscard patch is only necessary on Android API <21 since the missing defines are now properly included in the NDK

osm0sis committed:
Bump to 1.25.1
- all continues to compile correctly with my changes

osm0sis committed:
patch: strings, offset format should be decimal not octal

osm0sis committed:
Revert "patch: strings, offset format should be decimal not octal"

- superceded by -t radix support is in 1.26.1

This reverts commit 272bc58a78797757bf504d0c2fdd5abac0ab5acc.

osm0sis committed:
fix: rebase hush patch for busybox 1.26.1

osm0sis committed:
fix: rebase ether_ntoa-udhcpd for busybox 1.26.1

osm0sis committed:
Remove Cygwin patches no longer needed
- host warnings "array subscript has type 'char'" no longer occur

osm0sis committed:
patch: fix new build warning on Cygwin when using menuconfig

osm0sis committed:
patch: ipneigh/iproute/iprule, add missing headers (API <21)

osm0sis committed:
Revert "patch: ipneigh, do not include neighbour.h on Android"
- superceded by ipneigh/iproute/iprule headers patch

This reverts commit 3843d7a463e4d829b06d172ccb4b76f1fb740aa4.

osm0sis committed:
Bringup to busybox 1.26.2 with updated configs
- nandwrite and nanddump now cannot be built by default on all Android API since they now reference MTD_FILE_MODE_RAW (https://git.busybox.net/busybox/commit/miscutils/nandwrite.c?id=76915bf738c4532c7ca57fc673b5a0ebd4b91af8); the existing patch (originally only for API 21) resolves this and should now be used for all API.
- iproute and iprule now cannot be built by default on Android API <21 since they now reference RTA_TABLE (https://git.busybox.net/busybox/commit/networking/libiproute/iproute.c?id=b42107f21538e39d9a344376372f8261aed589b2); a new headers patch has replaced the ipneigh patch (since neighbour.h was also required) to resolve this.

osm0sis committed:
patch: add missing defines for x86 and MIPS NDK
- thanks to @ukanth for researching and submitting the diff for patch creation (see #1)
- fixes cross-compile issues that previously required ipcrm/ipcs + syslogd/logread to be disabled for these platforms
- caveat can now be removed from full config, applet list is finally uniform for 32-bit platforms

osm0sis committed:
cosmetic; change new patch to match others

osm0sis committed:
patch: work around several broken wget/ping/nslookup/etc resolver issues

osm0sis committed:
fix: rearrange to group new networking patches

osm0sis committed:
patch: workaround dns issues by forcing known dns for now

osm0sis committed:
config/README updates
- disable REBOOT and POWEROFF in basic configs since they do not function correctly on Android

osm0sis committed:
fix cosmetic typo

osm0sis committed:
README: fix sublists

osm0sis committed:
patch: fix remaining issues with ipv6 on Android API 21+

restores full ipv6 building on all 64-bit architectures

osm0sis committed:
patch: fix time and hush on Android API 21+ and 18+ respectively

osm0sis committed:
config/README fixes and updates
- CONFIG_TIME cannot be built by default on Android API >21 since wait3 was removed from bionic, but leave enabled in 32-bit configs since 32-bit builds will still often use a lower API level than that
- similarly, the issetugid stub was removed from bionic for Android API >18, so add it back as well since our patched CONFIG_HUSH needs it
- new IPV6 patch allows enabling all IPV6 + ROUTE utils on API >21, and now CONFIG_SYSLOGD/CONFIG_LOGREAD + CONFIG_IPC* appear to work again
- 32-bit and 64-bit full configs are now identical, but a different patchset is required, so keep them distinct for the patch list

osm0sis committed:
patch: fix reboot and poweroff

osm0sis committed:
config: update patchset for full 64-bit

osm0sis committed:
Revert "patch: workaround dns issues by forcing known dns for now"
- to be superceded by newer workaround patch

This reverts commit ca23adaecca8ea15723f08771956b857ccd7eeaf.

osm0sis committed:
patch: fix dietlibc dns resolver issues

osm0sis committed:
patch: fix udhcpc6, work around NDK missing ifaddrs

osm0sis committed:
Bringup to busybox 1.27.1 with updated configs
- udhcpc6 now causes missing header and then undefined reference errors due to the inclusion of ifaddrs.h and associated functions (https://git.busybox.net/busybox/commit/networking/udhcp/d6_socket.c?id=e09f5e3045fc90547be9ec49c63b633d103cfc45); a new ipv6 patch adds these functions and definitions that the NDK is missing

osm0sis committed:
patch: fix tls on x86, work around erroneous define assumption

osm0sis committed:
Update README

osm0sis committed:
Bump to 1.27.2
- all continues to compile correctly with my changes

osm0sis committed:
fix: rebase minix patch for busybox 1.28.1

osm0sis committed:
fix: rebase time patch for busybox 1.28.1

osm0sis committed:
fix: rebase ash patch for busybox 1.28.1

osm0sis committed:
patch: fix df, work around missing header (API <21)

osm0sis committed:
patch: fix build + run-init/setpriv capabilities, replace old headers (API <21)

osm0sis committed:
Bringup to busybox 1.28.1 with updated configs
- df now causes missing header errors on Android API <21 due to the inclusion of statvfs.h (https://git.busybox.net/busybox/commit/?id=d1535216ca27047e3962d61b975bd2a638aa45a2); a new df patch adds a dummy header to wrap the definitions and struct back to the working one
- run-init/setpriv capabilities introduced declaration errors on Android API <21 due to the addition of a number of _LINUX_CAPABILITY and PR_CAPBSET definitions that aren't available (https://git.busybox.net/busybox/commit/?id=200bcc851acbe1ba30fe90b5cf918f88370a5d15); a new runinit/setpriv patch replaces the older headers locally to provide the definitions required
- setfattr introduced missing header and then undefined reference errors on Android API <21 due to the use of xattr headers and functions not available in bionic libc with that version (https://git.busybox.net/busybox/commit/?id=b278ac0e607aa971c46706be63c580a2fd88dd56); these are ASM functions and thus extremely difficult to add, so disable on 32-bit configs since 32-bit builds will still often use a lower API level than that
- disable CONFIG_TLS on basic config per previously documented patch requirement on Android x86

The run-init issue is actually so severe it breaks basic unpatched compilation using Android NDK API <21 (https://bugs.busybox.net/show_bug.cgi?id=10791), so suggest reversion of those commits as a simple workaround until it can be resolved

osm0sis committed:
Bump to 1.28.2
- all continues to compile correctly with my changes
- add the omitted version bump to custom Makefile patch

osm0sis committed:
Bump to 1.28.3
- all continues to compile correctly with my changes
- revert to usual custom Makefile patch
- remove unnecessary .gitignore

osm0sis committed:
Bump to 1.28.4
- all continues to compile correctly with my changes

osm0sis committed:
fix: rebase dietlibc resolver/nslookup patch for busybox 1.29.1

osm0sis committed:
patch: fix ftpd/pw_encrypt

osm0sis committed:
patch: fix chrt (API <21)

osm0sis committed:
patch: fix tc (API <21)

osm0sis committed:
Bringup to busybox 1.29.1 with updated configs
- ftpd (among others) now causes missing header errors due to the addition of a glibc header bionic doesn't have (https://git.busybox.net/busybox/commit/?id=57dbe4d5ac483f928b6e80b33993c8fd6a061aa7); a new ftpd/pw_encrypt patch avoids including this on Android
- chrt introduced declaration errors on Android API <21 due to the addition of a SCHED_BATCH definition which isn't available (https://git.busybox.net/busybox/commit/?id=1616fb8aca23fd7ba6b379b0fb38d3327d1e31d1); a new chrt patch adds this definition when not already defined
- disable CONFIG_SSL_CLIENT and CONFIG_FEATURE_WGET_HTTPS on basic config since they enable CONFIG_TLS, which needs to be disabled per previously documented patch requirement on Android x86
- new default applet CONFIG_TC doesn't build on Android API <21 due to a missing struct member; a new patch replaces the older headers locally to provide the updated struct
- capabilities compilation issue on Android API <21 (https://bugs.busybox.net/show_bug.cgi?id=10791) is now fixed, and can be avoided by disabling CONFIG_RUN_INIT and CONFIG_FEATURE_SETPRIV_CAPABILITIES + CONFIG_FEATURE_SETPRIV_CAPABILITY_NAMES
- disable new internal resolver CONFIG_FEATURE_NSLOOKUP_BIG and CONFIG_FEATURE_NSLOOKUP_LONG_OPTIONS due to numerous declaration errors on Android API <21 and not working out-of-the-box on >21
- correct typo in 050-b-xconnect-dietlibc.patch description

osm0sis committed:
Bump to 1.29.2
- all continues to compile correctly with my changes

osm0sis committed:
update configs for Android Pie
- disable CONFIG_FEATURE_SUID to fix "Bad system call" errors from Android Pie's seccomp (thanks @topjohnwu)

osm0sis committed:
disable monotonic syscall on basic configs
- at least x86 on <API 21 has issues with clock_gettime(MONOTONIC) errors with, e.g., wget

osm0sis committed:
make intended Android API for NDK compiles more clear in configs

osm0sis committed:
Bump to 1.29.3
- all continues to compile correctly with my changes

osm0sis committed:
Prepare for NDK Unified Headers support
- drop all patches and configs for API <21 (doesn't matter anyway since we compile static)
- rearrange to better group remaining dependent patches
- rename some patches to prepare for splitting off extra portions which the Unified Headers don't require

osm0sis committed:
Update patches for NDK Unified Headers support
- split some patches (semctl, ipcs, syslogd) to avoid sem.h path fix needed for Deprecated Headers breaking compilation with Unified Headers
- add fixes for FAST_FUNC and glob now needed with Unified Headers (thanks topjohnwu)

osm0sis committed:
Update README/configs
- add NDK Unified Headers configs, which allows support of nano timestamps with date and requires fewer patches
- update NDK Deprecated Headers full config for new patchset
- enable ash as sh in all configs
- clarify poweroff/reboot don't "entirely" work, since they do work out-of-the-box with -f to force

osm0sis committed:
Restore erroneously removed TLS patch

osm0sis committed:
Cleanups after auditing patches for NDK Deprecated Headers

osm0sis committed:
Cosmetic patch changes

osm0sis committed:
Update README/configs, add full errorlists
- cleanups and updates following old patch purge and updated target API level
- add errorlist logs for full documentation of all associated errors with each full config
- seems like df, mount, etc. also don't work correctly, though build without error
- tls montgomery and time are an issue on basic unified config as well

osm0sis committed:
Update basic configs for broken mntent utils
- these all build fine now with API >21 but don't run correctly (no output, generally)

osm0sis committed:
Enable unicode properly in all configs

osm0sis committed:
fix: rebase ash history patch for busybox 1.30.1

osm0sis committed:
Bringup to busybox 1.30.1 with updated configs
- new bc utility does not build due to a code error

osm0sis committed:
Enable unshare in full configs

osm0sis committed:
Update README/configs
- disable broken network utils in basic configs since these don't entirely work from NDK static binary compiles of busybox without patches ("bad address" error with any attempt to resolve a hostname)

osm0sis committed:
fix: rebase xconnect/dietlibc patch for busybox 1.31.1

osm0sis committed:
Bringup to busybox 1.31.1 with updated configs

osm0sis committed:
Enable mtd-utils flash_* applets in full configs, update README

osm0sis committed:
Update README

osm0sis committed:
Add full config with SELinux support

osm0sis committed:
Update SELinux config with necessary fixes

osm0sis committed:
patch: work around broken install command with SELinux enabled
Thanks topjohnwu!

osm0sis committed:
Remove upstreamed patches for busybox 1.32.1

osm0sis committed:
fix: rebase hush/glob patch for busybox 1.32.1

osm0sis committed:
patch: fix missing librt error for busybox 1.32.1

osm0sis committed:
Bringup to busybox 1.32.1 with updated configs/errorlists

- a guard to avoid the glibc crypt.h include was upstreamed, so the ftpd/pw_encrypt patch is no longer necessary
- librt is now part of the build, so no basic config can build on older NDK where librt is not available (https://git.busybox.net/busybox/commit/?id=be5a505d771a77c640acc35ceaa470c80e62f954); a new Makefile.flags patch tests for librt availability and modifies the linked libraries accordingly

osm0sis committed:
Revert "Remove Cygwin patches no longer needed"

This reverts commit f86307349514dcd0c9830931d0ae77603b9b10b0.

Some of these are back with a vengeance in busybox 1.32.1

osm0sis committed:
Remove Cygwin patches that are still no longer needed, rearrange

osm0sis committed:
Combine remaining Cygwin host warning patches, rearrange

osm0sis committed:
Update errorlists and config patchlists

osm0sis committed:
fix: better gcc-version.sh patch for output spam issue

osm0sis committed:
fix: add better tls patch from upstream for now

https://bugs.busybox.net/show_bug.cgi?id=13716

It will be part of the currently unreleased busybox 1.34

osm0sis committed:
fix: rebase mount/umount/fsck/df patch for busybox 1.33.1

osm0sis committed:
fix: rebase modinfo/modprobe patch for busybox 1.33.1

osm0sis committed:
Bringup to busybox 1.33.1 with updated configs/errorlists

osm0sis committed:
fix: rebase ipv6 interface/route patch for busybox 1.33.1

osm0sis committed:
fix: proper rebase of modinfo/modprobe patch for busybox 1.33.1

osm0sis committed:
fix: correct dietlibc/dnscruft patch comment

osm0sis committed:
fix: rebase ash history patch for busybox 1.33.1

osm0sis committed:
Update README

- add base32 to default list
- note build will fail with no patches currently

osm0sis committed:
fix: update ash history patch for modern booted Android

- fixes flock -c
 https://android.googlesource.com/platform/bionic/+/refs/tags/android-11.0.0_r39/tests/grp_pwd_test.cpp#81

Thanks @vvb2060

osm0sis committed:
Remove upstreamed patches for busybox 1.34.1

osm0sis committed:
fix: rebase hush-platform patches for busybox 1.34.1

- time/wait3 hunk of patches was upstreamed, combine remaining hush-platform patches and rebase
- group all hush patches together

osm0sis committed:
Bringup to busybox 1.34.1 with updated configs/errorlists

osm0sis committed:
Update README

- librt fix has been merged

osm0sis committed:
Create FUNDING.yml

osm0sis committed:
Update README.md

- time wait3 and TLS ASM patches were upstreamed in 1.34.x

adligit2 commented:
> git remote add origin https://github.com/seifoh/seifoo.git git push -u origin master

adligit2 reviewed (approved):
add patch

adligit2 reviewed (approved):
add patch

adligit2 commented:
merge

Wizardboy92 commented:
Good

Wizardboy92 reviewed (changes requested):
Good

osm0sis committed:
Bringup to busybox 1.36.1 with updated configs

- powertop now fails with an asm error on x86 with NDK r15c (last with MIPS support), so disable for all configs

osm0sis committed:
fix: rebase ash history patch for busybox 1.36.1

Commit:
Readme Nitpick: cd (#2)

osm0sis committed:
patch: fix direct loopback mount and losetup next loop

osm0sis committed:
Update configs

osm0sis committed:
Update configs

- fix loop mounts on kernels < 5.x; TRY is supposed to fallback gracefully, but something doesn't work correctly, even with my new patch, so update config options to revert to old behavior

osm0sis committed:
Update README.md
